### PR TITLE
Fix issue 5039

### PR DESCRIPTION
**Pull request by dominikl**
# What this PR does

Fixes https://github.com/openmicroscopy/openmicroscopy/issues/5039

# Testing this PR

Check integration tests and code review.

### CODE REVIEW

File: components/rendering/test/util/mocks/MethodSignature.java [hunk 1]
```diff
@@ -242,8 +242,10 @@ public boolean equals(Object methodSignature) {
         boolean b = methodSignature != null && methodSignature.getClass() == MethodSignature.class;
         if (b) {
             MethodSignature ms = (MethodSignature) methodSignature;
+            String nameNotNull = name == null ? "" : name;
+            String msNameNotNull = ms.name == null ? "" : ms.name;
```
Comment (mtbc) [inline components/rendering/test/util/mocks/MethodSignature.java]: might want a look at `ObjectUtils.equals`

File: components/server/src/ome/security/SmtpAuthenticator.java [hunk 2]
```diff
@@ -17,15 +17,16 @@
 
     public SmtpAuthenticator(String username, String password) {
         super();
-        if (!"".equals(username) && !"".equals(password)) {
+        if (username != null && !username.isEmpty() && password != null
```
Comment (mtbc) [inline components/server/src/ome/security/SmtpAuthenticator.java]: `StringUtils.isNotEmpty` may help

**Comment by dominikl** [comment]
I'd use them if it were Insight, but I was afraid of maybe creating new dependencies in the other components, that's why I used `== null || isEmpty()`.

**Comment by mtbc** [comment]
I think we're okay with the basic Apache utils and Guava classes in OMERO.server; probably worth it for readability but happy to brook dissent.

**Review by mtbc — APPROVED** [review]

File: components/server/src/ome/security/SmtpAuthenticator.java [hunk 3]
```diff
@@ -10,22 +10,24 @@
...
+        if (StringUtils.isNotEmpty(username) && StringUtils.isNotEmpty(password)) {
             this.username = username;
             this.password = password;
         }
     }
 
     @Override
     public PasswordAuthentication getPasswordAuthentication() {
-        if (this.username != "" && this.password != "") {
+        if (!this.username.isEmpty() && !this.password.isEmpty()) {
```
Comment (jburel) [inline components/server/src/ome/security/SmtpAuthenticator.java:30]: another alternative is ``StringUtils.isNotEmpty()`` but it is fine as it is

**Pull request merged**